File: src/cli/update_command.zig
pub const UpdateCommand = struct {
    pub fn exec(ctx: Command.Context) !void {
        const cli = switch (try PackageManager.CommandLineArguments.parse(ctx.allocator, .update)) {
            .args => |a| a,
            .err => |f| bun.install.InstallResult.exitForCli(f),
        };

        if (cli.interactive) {
            const UpdateInteractiveCommand = @import("./update_interactive_command.zig").UpdateInteractiveCommand;
            try UpdateInteractiveCommand.exec(ctx);
        } else {
            (try updatePackageJSONAndInstallCatchError(ctx, .update)).handleCli();
        }
    }
};

const bun = @import("bun");
const Command = bun.cli.Command;

const PackageManager = bun.install.PackageManager;
const updatePackageJSONAndInstallCatchError = PackageManager.updatePackageJSONAndInstallCatchError;

File: src/runtime/cli/update_command.rs
use bun_install::package_manager_real::Subcommand;
use bun_install::package_manager_real::command_line_arguments::CommandLineArguments;

use crate::cli::pm_update_package_json::update_package_json_and_install_catch_error;
use crate::cli::update_interactive_command::UpdateInteractiveCommand;
use crate::command::Context;

pub(crate) struct UpdateCommand;

impl UpdateCommand {
    pub(crate) fn exec(ctx: Context) -> Result<(), crate::Error> {
        let cli = CommandLineArguments::parse(Subcommand::Update)?;

        if cli.interactive {
            UpdateInteractiveCommand::exec(ctx)?;
        } else {
            update_package_json_and_install_catch_error(ctx, Subcommand::Update)?;
        }
        Ok(())
    }
}
